crates/dbs-virtio-devices/src/fs/handler.rs

```diff
@@ -495,19 +495,99 @@ where
 pub mod tests {
     use std::sync::Arc;
 
+    use crate::fs::device::tests::*;
+    use crate::fs::*;
+    use crate::tests::{VirtQueue, VIRTQ_DESC_F_NEXT, VIRTQ_DESC_F_WRITE};
+    use crate::VirtioQueueConfig;
     use dbs_interrupt::NoopNotifier;
     use dbs_utils::epoll_manager::EpollManager;
     use dbs_utils::epoll_manager::SubscriberOps;
     use dbs_utils::rate_limiter::TokenBucket;
     use vm_memory::{GuestAddress, GuestMemoryMmap};
-
-    use crate::fs::device::tests::*;
-    use crate::fs::*;
-    use crate::tests::{VirtQueue, VIRTQ_DESC_F_NEXT, VIRTQ_DESC_F_WRITE};
-    use crate::VirtioQueueConfig;
+    use vmm_sys_util::tempfile::TempFile;
 
     use super::*;
 
+    #[test]
+    fn test_is_req_valid() {
+        let handler = CacheHandler {
+            cache_size: 0x1000,
+            mmap_cache_addr: 0x1000,
+            id: "test".to_string(),
+        };
+
+        // Normal case.
+        assert!(handler.is_req_valid(0x0, 0x500));
+
+        // Invalid case.
+        assert!(!handler.is_req_valid(0x500, 0x1000));
+    }
+
+    #[test]
+    fn test_map() {
+        let mut handler = CacheHandler {
+            cache_size: 0x10000,
+            mmap_cache_addr: 0x10000,
+            id: "test".to_string(),
+        };
+
+        // Normal case.
+        let file = TempFile::new().unwrap().into_file();
+        file.set_len(0x1000).unwrap();
+        let fd = file.as_raw_fd();
+        assert!(handler.map(0x0, 0x5000, 0x5000, 0, fd).is_ok());
+
+        // Invalid argument case.
+        assert!(matches!(
+            handler
+                .map(0x0, 0x5000, 0xc000, 0, fd)
+                .err()
+                .unwrap()
+                .kind(),
+            std::io::ErrorKind::InvalidInput
+        ));
+
+        // Bad file descriptor case.
+        let fd = TempFile::new().unwrap().as_file().as_raw_fd();
+        assert!(format!(
+            "{:?}",
+            handler.map(0x0, 0x5000, 0x5000, 0, fd).err().unwrap()
+        )
+        .contains("Bad file descriptor"));
+    }
+
+    #[test]
+    fn test_unmap() {
```

Comment:
Can you add a case that do unmap after map?

Reply:
When map() done, you can check the file's content is mapped to the specific address or not, and when unmap() done, you can check the specific address's content is none or not